sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -180,6 +221,46 @@ case class Join(
     }
   }
 
+  private def constructIsNotNullConstraints(condition: Expression): Set[Expression] = {
```

Comment:
Why is this done as a special case here, instead of doing it as part of `getRelevantConstraints`?